tests/meta-mender-ci/recipes-testing/mender-coverage-client/mender-client_%.bbappend

```diff
@@ -0,0 +1,67 @@
+SUMMARY = "Build a Mender client which records coverage during its invocation"
+
+DEPENDS += "gobinarycoverage rsync-native"
+
+do_copy_src_to_build[nostamp] = "1"
+do_copy_src_to_build[doc] = "Copies the Mender client source from ${S}/src/${GO_IMPORT} \
+to ${B}/src/${GO_IMPORT}"
+do_copy_src_to_build[dirs] = "${B}src/github.com/mendersoftware/mender"
+do_copy_src_to_build () {
+    bbwarn "Copying source to the build dir..."
+    rsync -a ${S}/src/${GO_IMPORT}/ ${B}/src/${GO_IMPORT}/
+    cd ${B}/src/${GO_IMPORT}/
+    # Why is this needed?
+    git reset --hard HEAD
+}
+
+do_instrument_client[dirs] =+ "${GOTMPDIR}"
+do_instrument_client[doc] = "Modifies the Mender client source to enable coverage analysis"
+do_instrument_client () {
+    # Since the Makefile installs its dependent tools using 'go get', add
+    # ${B}/bin to the path during the compilation
+    PATH="${B}/bin:$PATH"
+    export PATH
+    cd ${B}/src/${GO_IMPORT}
+    bbwarn "Building the Mender client with coverage analysis"
+    oe_runmake instrument-binary
+}
+
+python () {
+    # Only add coverage analysis if the client is newer than 2.2.x
+    if not mender_is_2_2_or_older(d):
+        bb.warn("Building the client with coverage instrumentation enabled")
+        # Disable do_configure inherited from go.bbclass. This function tries to
+        # symlink to the builddir from source i.e., ln -s ${S}/src ${B}/src.
+        d.setVarFlag('do_configure', 'noexec', "1")
+        # Copy the source over to the build directory instead of symlinking
+        # so that the source can be dirtied freely
+        # addtask do_copy_src_to_build before do_compile after do_configure
+        bb.build.addtask('do_copy_src_to_build', 'do_compile', 'do_configure', d)
+        # Coverage instrument the client before compiling it
+        d.prependVarFlag('do_compile', 'prefuncs', "do_instrument_client ")
```

Comment:
Why not do `addtask` here too?

Reply:
Because I dont want a separate stamp for it, it is directly related to re-compilation